packages/common/src/visualizations/SqlResultsRunner.ts

```diff
@@ -337,128 +339,30 @@ export class SqlRunnerResultsTransformer
-            : undefined;
-
-        const metricId = firstNumericColumn?.reference || undefined;
-
-        if (
-            !groupFieldIds ||
-            groupFieldIds.length === 0 ||
-            metricId === undefined
-        ) {
-            return undefined;
-        }
-
-        return {
-            groupFieldIds,
-            metricId,
-        };
-    }
-
-    public async getPieChartData(): // config: SqlTransformPieChartConfig,
-    Promise<PieChartData> {
-        // TODO: NYI
-        return {
-            results: this.rows,
-        };
-    }
-
-    // TODO: merge with function below
-    getPieChartConfig({
+    mergePivotChartLayout({
         currentConfig,
     }: {
-        currentConfig?: PieChartSqlConfig;
-    }) {
-        const newDefaultConfig = this.defaultPieChartLayout();
-
-        let newConfig = currentConfig;
-
-        if (!currentConfig) {
-            newConfig = {
-                metadata: {
-                    version: 1,
-                },
-                type: ChartKind.PIE,
-                fieldConfig: newDefaultConfig,
-                display: {
-                    isDonut: false,
-                },
-            };
-        } else {
-            newConfig = {
-                ...currentConfig,
-                fieldConfig: newDefaultConfig,
-            };
-        }
-
-        return {
-            newConfig,
-        };
-    }
-
-    // TODO: rename
-    getChartConfig({
-        chartType,
-        currentConfig,
-    }: {
-        chartType: ChartKind.VERTICAL_BAR | ChartKind.LINE;
-        currentConfig?: CartesianChartSqlConfig;
+        currentConfig?: SqlPivotChartLayout;
     }) {
         const newDefaultLayout = this.defaultPivotChartLayout();
 
         const someFieldsMatch =
-            currentConfig?.fieldConfig?.x.reference ===
-                newDefaultLayout?.x.reference ||
+            currentConfig?.x.reference === newDefaultLayout?.x.reference ||
             intersectionBy(
-                currentConfig?.fieldConfig?.y || [],
+                currentConfig?.y || [],
                 newDefaultLayout?.y || [],
                 'reference',
             ).length > 0;
 
-        let newConfig = currentConfig;
-
-        if (!currentConfig) {
-            newConfig = {
-                metadata: {
-                    version: 1,
-                },
-                type: chartType,
-                fieldConfig: newDefaultLayout,
-                display: undefined,
-            };
-        } else if (currentConfig && !someFieldsMatch) {
-            newConfig = {
-                ...currentConfig,
-                fieldConfig: newDefaultLayout,
-            };
+        let mergedLayout = currentConfig;
+
+        if (!currentConfig || !someFieldsMatch) {
+            mergedLayout = newDefaultLayout;
         }
 
         return {
-            newConfig,
-            newDefaultLayout,
+            defaultLayout: newDefaultLayout,
```

Comment:
is this `mergedLayout` thing needed? Still unsure why. Either way, it's now generic to cartesian + pie

Reply:
I think the merged layout is good because it keeps us from totally replacing the config when some fields match. 

BUT, I don't think we actually use defaultLayout anymore. It gets stored in redux after this return and seemingly never read again. Want to remove it? I could also take a not to follow up and look closer.